app/Client/AdPayClient.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * Copyright (c) 2018 Adshares sp. z o.o.
+ *
+ * This file is part of AdServer
+ *
+ * AdServer is free software: you can redistribute and/or modify it
+ * under the terms of the GNU General Public License as published
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * AdServer is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty
+ * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with AdServer. If not, see <https://www.gnu.org/licenses/>
+ */
+
+declare(strict_types = 1);
+
+namespace Adshares\Adserver\Client;
+
+interface AdPayClient
+{
+    public function updateCampaign(array $campaigns): void;
+
+    public function deleteCampaign(array $campaignIds): void;
+
+    public function addEvents(array $events): void;
+
+    public function getPayments(int $timestampFrom, int $timestampTo): array;
+}
```

Comment:
I'd move this one interface into our Domain :) (src/Demand/Application/Service/AdPay.php)
In the same way, we use AdSelect Client interface For Supply context.
Then we have only our JsonRpc implementation in app/ layer.
Sounds ok?

Reply:
OK